### PR TITLE
Migration to yeoman 1.0

### PR DESCRIPTION
**Pull request by pascalgrimaud**
Fix https://github.com/jhipster/generator-jhipster-module/issues/14

### CODE REVIEW

**Comment by deepu105** [comment]
You could also update the eslint rules to be same as jhipster

**Comment by pascalgrimaud** [comment]
Yes, of course!
I'm not finished yet, that's why I put [WIP] in the title

File: generators/app/templates/_gulpfile.js [hunk 1]
```diff
@@ -1,102 +1,95 @@
...
-  var sequence = require('gulp-sequence');
-  var path = require('path');
-  var eslint = require('gulp-eslint');
-  var excludeGitignore = require('gulp-exclude-gitignore');
-  var mocha = require('gulp-mocha');
-  var istanbul = require('gulp-istanbul');
-  var nsp = require('gulp-nsp');
-  var plumber = require('gulp-plumber');
+gulp.task('eslint', () => gulp.src(['gulpfile.js', 'generators/app/index.js', 'test/*.js'])
+    // .pipe(plumber({errorHandler: handleErrors}))
```
Comment (pascalgrimaud) [inline generators/app/templates/_gulpfile.js]: I let this line commented, as I don't know how to handle it

**Comment by pascalgrimaud** [comment]
@deepu105 : I know, it's a lot of files to review (sorry), but can you have a look when you have time plz? (it's not urgent).
If you don't have time, at least, if you can test the hook mechanism, as it's the most complex part for me.

Big thanks in advance.

**Comment by pascalgrimaud** [comment]
I don't understand why the coverage is still à 0% :cry: 
```
-----------|----------|----------|----------|----------|----------------|
File       |  % Stmts | % Branch |  % Funcs |  % Lines |Uncovered Lines |
-----------|----------|----------|----------|----------|----------------|
 app/      |        0 |        0 |        0 |        0 |                |
  index.js |        0 |        0 |        0 |        0 |... 195,196,197 |
-----------|----------|----------|----------|----------|----------------|
All files  |        0 |        0 |        0 |        0 |                |
-----------|----------|----------|----------|----------|----------------|
```

**Comment by pascalgrimaud** [comment]
I installed istanbul globally, then run `istanbul cover node_modules/.bin/_mocha` (thanks @windkomo)
Here the coverage:
```
=============================================================================
Writing coverage object [/home/pgrimaud/projects/jhipster/generator-jhipster-module/coverage/coverage.json]
Writing coverage reports at [/home/pgrimaud/projects/jhipster/generator-jhipster-module/coverage]
=============================================================================

=============================== Coverage summary ===============================
Statements   : 94.03% ( 63/67 )
Branches     : 66.67% ( 12/18 )
Functions    : 100% ( 5/5 )
Lines        : 95.45% ( 63/66 )
================================================================================
```

So there is something wrong in my gulpfile.js

**Comment by pascalgrimaud** [comment]
Found the issue: https://github.com/sindresorhus/gulp-mocha/issues/159
So I will downgrade gulp-mocha to 3.0.1

**Pull request merged**